test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add function call in function body"
+		);
+	}
+);
+
+// Remove function call from a function's body
+static YPR<FunctionDef> removeFuncCallInFuncBody(
+	[](FunctionDef* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<FunctionDef>(
+			[](FunctionDef* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_block()->mutable_statements())
+					if (stmt.has_functioncall())
+					{
+						delete stmt.release_functioncall();
+						stmt.clear_functioncall();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Remove function call from function body"
+		);
+	}
+);
+
+// Add dataoffset/datasize
+static YPR<Expression> addDataExpr(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				auto unopdata = new UnaryOpData();
+				unopdata->set_identifier(_rand());
+				unopdata->set_op(
+					YPM::EnumTypeConverter<UnaryOpData_UOpData>{}.enumFromSeed(_rand())
+				);
+				_message->set_allocated_unopdata(unopdata);
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Mutate expression to dataoffset/size"
+		);
+	}
+);
+
+// Add variable reference inside for-loop body
+static YPR<BoundedForStmt> addVarRefInForBody(
+	[](BoundedForStmt* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<BoundedForStmt>(
+			[](BoundedForStmt* _message, YulRandomNumGenerator& _rand)
+			{
+				auto popStmt = new PopStmt();
+				popStmt->set_allocated_expr(YPM::refExpression(_rand));
+				_message->mutable_for_body()->add_statements()->set_allocated_pop(popStmt);
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Add variable reference in for loop body"
+		);
+	}
+);
+
+// Mutate expression to a function call
+static YPR<Expression> mutateExprToFuncCall(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				auto functionCall = new FunctionCall();
+				functionCall->set_ret(FunctionCall::SINGLE);
+				functionCall->set_func_index(_rand());
+				_message->set_allocated_func_expr(functionCall);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate expression to function call"
+		);
+	}
+);
+
+// Mutate expression to variable reference
+static YPR<Expression> mutateExprToVarRef(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				_message->set_allocated_varref(YPM::varRef(_rand()));
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate expression to a variable reference"
+		);
+	}
+);
+
+// Add varref to statement
+static YPR<Statement> addVarRefToStmt(
```

Comment:
Yeah, `add_statements` adds protobuf messages of type `Statement` to a basic block i.e., `Block`. Here, we are mutating an existing statement message by adding variable references to it.

For example, the assignment statement looks like so

```
assignment {
  ref_id {
    varnum: 2
  }
  expr {}
}
```

- `ref_id` encodes the variable number to be referenced on the LHS (as in `0` in `x_0 := 2`)
- `expr` encodes the RHS (`2` in `x_0 := 2`)

Because of a limitation in libprotobuf-mutator, it can happen that either `expr` or `ref_id` or both are not initialized. I call `ref_id` and `expr` arguments of the statement.

This mutator `addVarRefToStmt` turns uninitialized arguments into variable references by initializing these arguments with the variable reference protobuf type.